File: C++/Threads.cpp
#include <iostream>
#include <thread>
#include <vector>
#include <cmath>

using namespace std;

// 9個のスレッドそれぞれで6回のループ
	// スレッドの内容を定義するこの関数をヘッダーファイルで定義するとコンパイルエラーになる
void mtAction(int m) {
	for (int n=1;n<7;n++) printf("|(%d,%d)| = %8.5f\n",m,n,std::hypot(m,n));
	// cout << << ... で出力する場合,値ごとにバラバラの出力になる
}

void Threads(){

	cout << "同時に実行可能な数: " << std::thread::hardware_concurrency() << endl;

	std::vector<std::thread> threads;

	for (int m=1;m<10;m++) threads.push_back(std::thread(mtAction,m));
	cout << "スレッドの終了を待ちます" << endl;
	for (std::thread &onethread : threads) onethread.join();
	cout << "スレッドは終了しました" << endl;

	cout << endl << endl;

}

File: PHP/Legacy/Threads.php
#! /usr/bin/env php
<?php

require_once("Utility.php");

function Threads() {
	global $i;

	pnl(1);

	println("これからスレッドを試します",2);

	# 9個のスレッドそれぞれで6回のループ

	class mtThread extends Thread {

		private $m=0;
		public function __construct($m) {
			$this->m = $m;
		}

		# 実行内容を定義
		public function run() {
			for ($n=1;$n<7;$n++) {
				print "|($this->m,$n)| = {$i(hypot($this->m,$n))}\r\n";
			}
		}

	}

	$threads=[];

	for ($m=1;$m<10;$m++) {
		array_push($threads,new mtThread($m));
	}

	println("スレッドを開始します");
	foreach ($threads as $t) {
		$t->start();
	}

	println("スレッドの終了を待ちます");
	foreach ($threads as $t) {
		$t->join();
	}
	println("スレッドは終了しました");

	# 各々のスレッド内でのループの順番は守られるが,全54回のプロセスの順番はバラバラ
	# このコードを実行する毎に,順番は変化する

	pnl(2);

};

if (running_directly(__FILE__)) Threads();

?>